portal/portal-service-impl/impl/src/java/org/sakaiproject/portal/service/PortalServiceImpl.java

```diff
@@ -618,5 +647,181 @@ public String getSkinPrefix() {
 		return "";
 	}
 
+	public List<BullhornAlert> getSocialAlerts(String userId) {
+
+		List<BullhornAlert> alerts = sqlService.dbRead(
+                "SELECT * FROM SOCIAL_ALERTS WHERE TO_USER = ? ORDER BY EVENT_DATE DESC"
+				, new Object[] { userId }
+				, new SqlReader() {
+						public Object readSqlResultRecord(ResultSet rs) {
+							return new BullhornAlert(rs);
+						}
+					}
+				);
+
+		for (BullhornAlert alert : alerts) {
+			try {
+				User fromUser = userDirectoryService.getUser(alert.from);
+				alert.fromDisplayName = fromUser.getDisplayName();
+			} catch (UserNotDefinedException unde) {
+				alert.fromDisplayName = alert.from;
+			}
+		}
+
+		return alerts;
+	}
+
+	public int getSocialAlertCount(String userId) {
+
+		Cache countCache = memoryService.newCache("bullhorn_alert_count_cache");
+
+		if (!countCache.containsKey(userId)) {
```

Comment:
Doing a contains and a get is a little inefficient. Just do the get, and check the return for null:

```
Map<String, Integer> cachedCounts = (Map<String, Integer>) countCache.get( userId );
if( cachedCounts == null )
{ ... }
```